jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -366,6 +366,8 @@ public boolean handle()
                 {
                     case TERMINATED:
                         onCompleted();
+                        _request.getHttpInput().failed(new IOException("terminated"));
+                        getEndPoint().cancelFillInterested();
```

Comment:
If we do create an exception for the step above, it should be passed into this cancel so we don't create another one.  Or maybe the HttpInput does this cancel for us?